src/sayt/autocomplete.ts

```diff
@@ -0,0 +1,25 @@
+import { QueryTimeAutocompleteConfig } from 'sayt';
+import { AutocompleteResultGroup } from './results';
+
+/** The name of the event fired when an autocomplete request is to be made. */
+export const AUTOCOMPLETE_REQUEST = 'sfx::autocomplete_request';
+/** The type of the [[AUTOCOMPLETE_REQUEST]] event payload. */
+export interface AutocompleteRequestPayload extends QueryTimeAutocompleteConfig {
+  /** The search term to autocomplete. */
+  query: string;
```

Comment:
How will the autocomplete response be made to include the `searchbox` if there is no `searchbox` in the request?

Reply:
I think that addressing this would be scope creep for this current ticket. There should be another task to determine that and in turn this interface can be updated as a part of it.

Reply:
I'll accept that, but I will strongly request adding the following task to your ticket: produce a list of current issues/incompatibilities/limitations/etc with the events in our application.

The problem is that we've been working on everything in isolation, and this is the first ticket to really work with everything. We therefore need this story to, at the very least, find flaws in our existing architecture. Does that seem reasonable?

Reply:
I believe the driver does not currently handle different search boxes as it is. Do we want to add it in now as an optional key so that this repo is ready when the logic layer is updated for that?

Reply:
I figured that adding in the searchbox would have been the responsibility of SFX-295, the ticket to have SAYT trigger the driver. This ticket was only for converting/normalizing what we already had, and since this functionality didn't already exist, we didn't add it in.

Reply:
That said, if it's just adding the property to the interface, I think we can absorb that in this ticket.